### PR TITLE
Feature/93

### LINKED ISSUE #93: Try to guess SMTP and IMAP connection settings
The app should be able to guess the settings of the server. If not able, show the user an advanced tab where insert custom settings

### PR DESCRIPTION
**Pull request by SimoPP**
Created the guess_imap() and guess_smtp() functions in app.py. 

They connect to the target server trying to find the port and protocol settings.
They return a dictionary with the settings

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -393,98 +394,88 @@ def get_email_platform():
...
+            connection = Imbox(
+                server,
+                username=user,
+                password=passwrd,
+                ssl=i[0],
+                starttls=i[1],
+                ssl_context=None
+            )
+            test = connection.connection.check()
+            print(test)
```
Comment (MattiaFailla) [inline app.py:429]: Please use Tentalog

Comment (MattiaFailla) [inline app.py]: Please use Tentalog

**Review by MattiaFailla — COMMENTED** [review]
Please add a delay between attempts to prevent the server from adding the client to the blocked ones. In addition, it would be better to migrate to Tentalog than to use the print statement.

File: app.py (hunk 1 above)
Comment (SimoPP) [reply app.py]: All corrected, thanks!

**Review by MattiaFailla — APPROVED** [review]

**Comment by MattiaFailla** [comment]
image

**Review by MattiaFailla — APPROVED** [review]

**Pull request merged**